In [2]:
import pandas as pd
import numpy as np
%matplotlib inline

In [3]:
# row 생략 없이 출력
pd.set_option('display.max_rows', None)
# col 생략 없이 출력
pd.set_option('display.max_columns', None)

In [4]:
df=pd.read_csv('../result_data/split_city.csv',encoding='utf8')
df.head()

Recv_Dept Nm       도     시 Recv_Cplt_Dm  Npa_Cl  Evt_CL_CD  Evt_Stat_Cd  \
0          충남청    충청남도   아산시   2021-05-03      19        501           10   
1          대전청   대전광역시   유성구   2021-05-03      13        601           10   
2          대전청   대전광역시   유성구   2021-04-27      13        201           10   
3          충남청    충청남도   보령시   2021-05-01      19        201           10   
4          충남청  [공중전화]  충청남도   2021-05-01      19        601           10   

       Hppn_X     Hppn_Y  Rpter_Sex Sme_Evt_Yn  
0  127.092762  36.781250          2          Y  
1  127.305689  36.332023          3          Y  
2  127.339578  36.347469          1          Y  
3  126.534663  36.242860          3          Y  
4  126.692715  36.168714          3          Y

### 월별 사건 사고 수 집계
### Recv Cplt Dm 의 값이 str 이라서 DataFrame.timestamp로 변형

In [31]:
df['Recv_Cplt_Dm'] = pd.to_datetime(df['Recv_Cplt_Dm'], format='%Y-%m-%d')
df.dtypes

Recv_Dept Nm            object
도                       object
시                       object
Recv_Cplt_Dm    datetime64[ns]
Npa_Cl                   int64
Evt_CL_CD                int64
Evt_Stat_Cd              int64
Hppn_X                 float64
Hppn_Y                 float64
Rpter_Sex                int64
Sme_Evt_Yn              object
dtype: object

## 전체 사건의 합을 count 했을 때 12, 9, 10, 8, 5 ... 순서로 사고가 많이 일어남

In [32]:
df_month_countof_total=df.groupby(df.Recv_Cplt_Dm.dt.month)['Recv_Cplt_Dm'].count().sort_values(ascending=False)

In [33]:
df_month_countof_total

Recv_Cplt_Dm
12    47807
9     42864
10    41893
8     39402
5     39374
7     39166
6     38840
11    38490
1     37013
4     33085
3     32000
2     25232
Name: Recv_Cplt_Dm, dtype: int64

## 총 사건의 발생 수 내림차순 순서

In [48]:
df_accident_countof_total=df.groupby(df.Evt_CL_CD)['Evt_CL_CD'].count().sort_values(ascending=False)

In [49]:
df_accident_countof_total

Evt_CL_CD
307    68892
501    41673
401    37322
301    32579
201    28731
601    25717
302    24833
210    19301
509    16520
406    15787
107    14234
604    14095
508    11575
305     9522
306     9461
402     9316
104     8779
503     6958
403     5872
211     5581
505     5221
215     4518
208     3595
308     3042
304     2908
303     2795
506     2623
206     2604
609     2384
205     2303
606     1974
603     1567
209     1337
108     1287
202     1241
502     1202
212     1100
106     1051
607     1018
207      926
602      787
204      608
213      526
610      339
105      307
507      234
216      226
214      214
109      186
504      157
101       50
102       35
203       19
605       13
404       10
103        6
405        5
Name: Evt_CL_CD, dtype: int64

### 월별 401~406 Dataframe 분류

In [28]:
df_transportation=df[(df['Evt_CL_CD']>400) & (df['Evt_CL_CD']<407)]

## 월별 401~406 각각 사건 수
## 401 406 402 403 404(5) 순서로 400번대 사건이 많이 일어남

In [40]:
df_month_countof=df_transportation.groupby(df_transportation.Recv_Cplt_Dm.dt.month)['Evt_CL_CD'].value_counts()

In [41]:
display(df_month_countof)

Recv_Cplt_Dm  Evt_CL_CD
1             401          3202
              406          1098
              402           738
              403           401
              405             3
2             401          2149
              406           740
              402           463
              403           260
3             401          2401
              406          1027
              402           550
              403           394
              405             1
4             401          2626
              406          1111
              402           578
              403           428
              404             1
5             401          3104
              406          1503
              402           670
              403           553
6             401          2869
              406          1388
              402           760
              403           535
              404             1
              405             1
7             401          3050
              40

## 월별 교통사고 수 합계

In [42]:
df_month_count_sum=df_transportation.groupby(df_transportation.Recv_Cplt_Dm.dt.month)['Evt_CL_CD'].count().sort_values(ascending=False).to_frame()

### 12 11 9 10 8 5 7 6 1 순서로 사고가 많이 일어남

In [43]:
display(df_month_count_sum)

Evt_CL_CD
Recv_Cplt_Dm           
12                 7858
11                 6514
9                  6424
10                 6310
8                  5907
5                  5830
7                  5744
6                  5554
1                  5442
4                  4744
3                  4373
2                  3612